src/opm/output/eclipse/AggregateActionxData.cpp

```diff
@@ -436,26 +450,60 @@ const std::map<cmp_enum, int> cmpToIndex = {
                     throw std::invalid_argument("Actionx: " + actx.name());
                 }
 
-                /*item [17] - relates to operator and if the right hand condition is a constant or not
+                /*item [17] - is an item that is non-zero for actions with several conditions combined using logical operators (AND / OR)
                  * First condition => [17] =  0
                  * Second+ conditions
-                    *If the previous condition has a constant rhs => [17] =  0
-                    *If rhs is {W,G, F} and
-                        *If previous condition is AND => [17] =  1
-                        *If previous condition is OR  => [17] =  0
+                    *Case - no parentheses
+                        *If all conditions before current condition has AND => [17] = 1
+                        *If one condition before current condition has OR   => [17] = 0
+                    *Case - parenthesis before first condition
+                        *If inside first parenthesis
+                            *If all conditions before current condition has AND [17] => 1
+                            *If one condition before current condition has OR  [17] => 0
+                        *If after first parenthesis end and outside parenthesis
+                            *If all conditions outside parentheses and before current condition has AND => [17] = 1
+                            *If one condition outside parentheses and before current condition has OR [17] => 0
+                        *If after first parenthesis end and inside a susequent parenthesis
+                            * [17] = 0
+                    *Case - parenthesis after first condition
+                        *If inside parentesis => [17] = 0
+                        *If outside parenthesis:
+                            *If all conditions outside parentheses and before current condition has AND => [17] = 1
+                            *If one condition outside parentheses and before current condition has OR [17] => 0
                 */
-                if (cond_it > actx_cond.begin()) {
-                    const std::string prev_rhs_quant = (cond_it-1)->rhs.quantity.substr(0,1);
-                    const auto it_prev_rhs = rhsQuantityToIndex.find(prev_rhs_quant);
-                    if (it_prev_rhs != rhsQuantityToIndex.end()) {
-                    const auto it_logic_17 = logicalToIndex_17.find((cond_it-1)->logic);
-                        if (it_logic_17 != logicalToIndex_17.end()) {
-                            iAcn[ind + 17] = it_logic_17->second;
-                        }
-                        else {
-                            std::cout << "Unknown Boolean operator type for condition: " << z_data.lhs.quantity << std::endl;
-                            throw std::invalid_argument("Actionx: " + actx.name());
-                        }
+                if (cond_it == actx_cond.begin()) {
```

Comment:
The logic for `Acn[ind + 17]` is quite complex. To simplify I suggest extracting it out of the main loop and creating a new loop over conditions only for `iAcn[ind + 17]` than can be prefixed with a size check:
```C++
if (actx_cond.size() > 1) {
     for (const auto& cond : actx_cond) {
          ....
     }
}
```

Reply:
I have extracted this loop as a separate loop and used the functions open_paren() and close_paren() to simplify the code. However, I need to do a special treatment of the first condition, hence I have kept the iterator for loop.